src/core/PlaneWaveBasis.jl

```diff
@@ -0,0 +1,176 @@
+using FFTW
+using StaticArrays
+
+struct PlaneWaveBasis{T <: Real}
+    # Lattice and reciprocal lattice vectors in columns
+    lattice::SMatrix{3, 3, T, 9}
+    recip_lattice::SMatrix{3, 3, T, 9}
+    unit_cell_volume::T
+    recip_cell_volume::T
+
+    # Selected energy cutoff at construction time
+    Ecut::T
+
+    # Size of the rectangular Fourier grid used as the density basis
+    # and the k-point-specific (spherical) wave function basis
+    # The wave vectors are given in integer coordinates.
+    grid_size::SVector{3, Int}
+    idx_DC::Int  # Index of the DC component
+    kpoints::Vector{SVector{3, T}}
```

Comment:
This is in fractional coordinates I guess?

Reply:
added a comment.